html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url-impact-on-history-pushreplacestate.window.js

```diff
@@ -0,0 +1,33 @@
+async_test(t => {
+  const frame = document.body.appendChild(document.createElement("iframe"));
+  t.add_cleanup(() => frame.remove());
+
+  frame.onload = t.step_func_done(() => {
+    // Per discussion in https://github.com/whatwg/html/issues/6556 and related threads, this will change the document's
+    // URL, but will *not* update the session history entry's URL.
+    frame.contentDocument.open();
+    assert_equals(frame.contentDocument.URL, document.URL, "Precondition: the URL gets changed");
+
+    frame.contentWindow.history.pushState(null, "");
```

Comment:
It took me a long time to realize that this passes null as URL. (I should maybe have read the commit message first.)

So the idea is that the fallback for `pushState()` is the document's URL.

Reply:
To be clear, the third parameter is the URL, and it's not passed at all.

Reply:
Right, but that defaults to null through IDL.